frappe/core/doctype/data_export/data_export.js

```diff
@@ -27,6 +27,9 @@ frappe.ui.form.on("Data Export", {
 			reset_filter_and_field(frm);
 		}
 	},
+	export_without_main_header: (frm) => {
+		frm.refresh();
+	}
```

Comment:
```suggestion
```

Not sure why this is required?

Reply:
Don't remember the issue now but it was most likely because the export button callback was not getting updated with the latest value for `frm` when the checkbox value changed.